File: Tp Soap/Q1/Q1/Hotel.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Collections;
namespace Q1
{
    class Hotel
    {

        private String nom;
        private Adresse adresse;
        private int nombreEtoile;
        private ArrayList chambres = null;

        public String getNom()
        {
            return nom;
        }
        public string Nom { get => nom; set => nom = value; }
        internal Adresse Adresse { get => adresse; set => adresse = value; }
        public int NombreEtoile { get => nombreEtoile; set => nombreEtoile = value; }
        public ArrayList Chambres { get => chambres; set => chambres = value; }

        public Hotel(String nom, Adresse adresse, int nombreEtoile)
        {
            this.Adresse = adresse; this.NombreEtoile = nombreEtoile; this.Chambres = new ArrayList();
            this.Nom = nom;

        }

        public Hotel() { }

        public Boolean addChambre(Chambre chambre)
        {
            if (Chambres.Add(chambre) >= 0)
            {
                return true;
            }
            else
            {
                return false;
            }
        }

    }
}

File: Tp Soap/Q1EnJava/src/Reserve/Hotel.java
package Reserve;

import java.util.ArrayList;

public class Hotel {

	public String nom;
	public Adresse adresse;
	public int nombreEtoile;
	public ArrayList<Chambre> chambres = null;
   // ArrayList< Chambre> chambres = null;


	public Hotel(String nom, Adresse adresse, int nombreEtoile) {

		this.nom = nom;
		this.adresse = adresse;
		this.nombreEtoile = nombreEtoile;
		this.chambres = new ArrayList();
	}

	public String getNom() {
		return nom;
	}

	public void setNom(String nom) {
		this.nom = nom;
	}

	public Adresse getAdresse() {
		return adresse;
	}

	public void setAdresse(Adresse adresse) {
		this.adresse = adresse;
	}

	public int getNombreEtoile() {
		return nombreEtoile;
	}

	public void setNombreEtoile(int nombreEtoile) {
		this.nombreEtoile = nombreEtoile;
	}

	public ArrayList getChambres() {
		return chambres;
	}

	public void setChambes(ArrayList chambres) {
		this.chambres = chambres;
	}

	public Hotel() {
	}
public Boolean addChambre(Chambre chambre){
	if(chambres.add(chambre)){
	return true;
	}
	return true;
}
}
